src/radical/pilot/agent/staging_input/default.py

```diff
@@ -199,7 +200,13 @@ def _handle_unit(self, unit, actionables):
                     self._log.error('no transfer for %s -> %s', src, tgt)
                     self._prof.prof('staging_in_fail', uid=uid, msg=did)
                     raise NotImplementedError('unsupported transfer %s' % src)
-
+            elif action == rpc.TARBALL:
+                # If somethig was staged via the tarball method, the tarball is
+                # extracted and then removed from the unit folder.
+                tar = tarfile.open(tgtdir+'/'+uid+'.tar')
```

Comment:
Per the `if` above, you can get `client://` type URLs here - do those need expansion?

Reply:
I do not think there is any need to check about `client:///`. The `if` in line 161 makes sure that whatever passes does not have `client://` at the target. Am I right?

I have a feeling that either I am missing something major,